packages/vite/src/node/__tests__/scan.spec.ts

```diff
@@ -123,3 +125,44 @@ describe('optimizer-scan:script-test', () => {
     expect(ret).not.toContain('export default')
   })
 })
+
+test('scan jsx-runtime', async (ctx) => {
+  const server = await createServer({
+    configFile: false,
+    logLevel: 'error',
+    root: path.join(import.meta.dirname, 'fixtures', 'scan-jsx-runtime'),
+    environments: {
+      client: {
+        // silence client optimizer
+        optimizeDeps: {
+          noDiscovery: true,
+        },
+      },
+      ssr: {
+        resolve: {
+          noExternal: true,
+        },
+        optimizeDeps: {
+          force: true,
+          noDiscovery: false,
+          entries: ['./entry-jsx.tsx', './entry-no-jsx.js'],
+        },
+      },
+    },
+  })
+
+  // TODO: should server listen be needed for optimizer?
```

Comment:
I think `server.listen` itself is not needed. The reason why it requires it now is probably:
- it is safer to start the optimizer after `buildStart` is called (https://github.com/vitejs/vite/pull/12832)
- it is safer to start the optimizer after the previous server is stopped (https://github.com/vitejs/vite/blob/557f797d29422027e8c451ca50dd84bf8c41b5f0/packages/vite/src/node/server/environment.ts#L184-L187)